File: sdks/python/bridgebank/models/classificar_transacoes_request.py
from __future__ import annotations
from collections.abc import Callable
from dataclasses import dataclass, field
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .transacao_dto import TransacaoDto

@dataclass
class ClassificarTransacoesRequest(AdditionalDataHolder, Parsable):
    # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additional_data: dict[str, Any] = field(default_factory=dict)

    # The transacoes property
    transacoes: Optional[list[TransacaoDto]] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: ParseNode) -> ClassificarTransacoesRequest:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: ClassificarTransacoesRequest
        """
        if parse_node is None:
            raise TypeError("parse_node cannot be null.")
        return ClassificarTransacoesRequest()
    
    def get_field_deserializers(self,) -> dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: dict[str, Callable[[ParseNode], None]]
        """
        from .transacao_dto import TransacaoDto

        from .transacao_dto import TransacaoDto

        fields: dict[str, Callable[[Any], None]] = {
            "transacoes": lambda n : setattr(self, 'transacoes', n.get_collection_of_object_values(TransacaoDto)),
        }
        return fields
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if writer is None:
            raise TypeError("writer cannot be null.")
        writer.write_collection_of_object_values("transacoes", self.transacoes)
        writer.write_additional_data_value(self.additional_data)
    

File: sdks/java/src/main/java/bridgebank/models/ClassificarTransacoesRequest.java
package bridgebank.models;

import com.microsoft.kiota.serialization.AdditionalDataHolder;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ClassificarTransacoesRequest implements AdditionalDataHolder, Parsable {
    /**
     * Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     */
    private Map<String, Object> additionalData;
    /**
     * The transacoes property
     */
    private java.util.List<TransacaoDto> transacoes;
    /**
     * Instantiates a new {@link ClassificarTransacoesRequest} and sets the default values.
     */
    public ClassificarTransacoesRequest() {
        this.setAdditionalData(new HashMap<>());
    }
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a {@link ClassificarTransacoesRequest}
     */
    @jakarta.annotation.Nonnull
    public static ClassificarTransacoesRequest createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new ClassificarTransacoesRequest();
    }
    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return a {@link Map<String, Object>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, Object> getAdditionalData() {
        return this.additionalData;
    }
    /**
     * The deserialization information for the current model
     * @return a {@link Map<String, java.util.function.Consumer<ParseNode>>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(1);
        deserializerMap.put("transacoes", (n) -> { this.setTransacoes(n.getCollectionOfObjectValues(TransacaoDto::createFromDiscriminatorValue)); });
        return deserializerMap;
    }
    /**
     * Gets the transacoes property value. The transacoes property
     * @return a {@link java.util.List<TransacaoDto>}
     */
    @jakarta.annotation.Nullable
    public java.util.List<TransacaoDto> getTransacoes() {
        return this.transacoes;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeCollectionOfObjectValues("transacoes", this.getTransacoes());
        writer.writeAdditionalData(this.getAdditionalData());
    }
    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public void setAdditionalData(@jakarta.annotation.Nullable final Map<String, Object> value) {
        this.additionalData = value;
    }
    /**
     * Sets the transacoes property value. The transacoes property
     * @param value Value to set for the transacoes property.
     */
    public void setTransacoes(@jakarta.annotation.Nullable final java.util.List<TransacaoDto> value) {
        this.transacoes = value;
    }
}
